language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -365,52 +454,180 @@ impl DependencyGraph {
+                    *to,
+                    parent,
+                    dep.clone(),
+                    skip_fetch_latest_git_deps,
+                    progress_output,
+                )?,
+            }
+        }
 
-            pkg.kind.reroot(&parent)?;
-            self.process_dependency(pkg, *to, skip_fetch_latest_git_deps, progress_output)?;
+        Ok(())
+    }
 
-            self.package_graph.add_edge(
-                from,
-                *to,
-                Dependency {
-                    mode: DependencyMode::DevOnly,
-                    subst: dep.subst.clone(),
-                    digest: dep.digest,
-                },
+    /// Resolve the packages described at dependency `to` of package `from` with manifest at path
+    /// `package_path` by running the binary `resolver.  `mode` decides whether the resulting
+    /// packages are added to `self` as dependencies of `package_name` or dev-dependencies.
+    ///
+    /// Sends progress updates to `progress_output`, including stderr from the resolver, and
+    /// captures stdout, which is assumed to be a lock file containing the result of package
+    /// resolution.
+    fn resolve_externally<Progress: Write>(
+        &mut self,
+        mode: DependencyMode,
+        from: PM::PackageName,
+        to: PM::PackageName,
+        resolver: Symbol,
+        package_path: &Path,
+        progress_output: &mut Progress,
+    ) -> Result<()> {
+        let mode_label = if mode == DependencyMode::DevOnly {
+            "dev-dependencies"
+        } else {
+            "dependencies"
+        };
+
+        let progress_label = format!("RESOLVING {} IN", mode_label.to_uppercase())
+            .bold()
+            .green();
+
+        writeln!(
+            progress_output,
+            "{progress_label} {to} {} {from} {} {resolver}",
+            "FROM".bold().green(),
+            "WITH".bold().green(),
+        )?;
+
+        // Call out to the external resolver
+        let output = Command::new(resolver.as_str())
+            .arg(format!("--resolve-move-{mode_label}"))
+            .arg(to.as_str())
+            .current_dir(package_path)
+            .output()
+            .with_context(|| format!("Running resolver: {resolver}"))?;
+
+        // Present the stderr from the resolver, whether the process succeeded or not.
+        if !output.stderr.is_empty() {
+            let stderr_label = format!("{resolver} stderr:").red();
+            writeln!(progress_output, "{stderr_label}")?;
+            progress_output.write_all(&output.stderr)?;
+        }
+
+        if !output.status.success() {
+            let err_msg = format!(
+                "'{resolver}' failed to resolve {mode_label} for dependency '{to}' of package \
+                 '{from}'"
             );
+
+            if let Some(code) = output.status.code() {
+                bail!("{err_msg}. Exited with code: {code}.");
+            } else {
+                bail!("{err_msg}. Terminated by signal.");
```

Comment:
Other `bail` messages don't finish with a period, so to keep things consistent probably best to remove them here.